exonum-java-binding-core/src/main/java/com/exonum/binding/transaction/Transaction.java

```diff
@@ -16,38 +16,20 @@
 
 package com.exonum.binding.transaction;
 
-import com.exonum.binding.common.hash.HashCode;
-import com.exonum.binding.common.message.BinaryMessage;
-import com.exonum.binding.storage.database.Fork;
-
 /**
  * An Exonum transaction.
  *
- * <p>You shall usually extend {@link AbstractTransaction} rather than implementing
- * this interface.
- *
  * @see <a href="https://exonum.com/doc/architecture/transactions">Exonum Transactions</a>
  * @see <a href="https://exonum.com/doc/architecture/services">Exonum Services</a>
  */
+@FunctionalInterface
 public interface Transaction {
 
-  /**
-   * Returns true if this transaction is valid: its data holds the invariants,
-   * it has a correct signature, etc.
-   *
-   * <p>This method is intended to check the <em>internal</em> consistency of a transaction.
-   * You shall <strong>not</strong> access any external objects in this method
-   * (e.g., files, network resources, databases).
-   *
-   * <p>If this method returns false, the transaction is considered incorrect,
-   * and Exonum discards it. Exonum never records invalid transactions into a blockchain.
-   */
-  boolean isValid();
-
   /**
    * Execute the transaction, possibly modifying the blockchain state.
    *
-   * @param view a database view, which allows to modify the blockchain state
+   * @param context a transaction execution context, which allows to access transaction-related
```

Comment:
which allows to access the information about this transaction and modify the blockchain state through the included database fork